Python/optimizer.c

```diff
@@ -1709,4 +1716,113 @@ _Py_Executors_InvalidateCold(PyInterpreterState *interp)
     _Py_Executors_InvalidateAll(interp, 0);
 }
 
+static void
+write_str(PyObject *str, FILE *out)
+{
+    // Encode the Unicode object to the specified encoding
+    PyObject *encoded_obj = PyUnicode_AsEncodedString(str, "utf8", "strict");
+    if (encoded_obj == NULL) {
+        PyErr_Clear();
+        return;
+    }
+    const char *encoded_str = PyBytes_AsString(encoded_obj);
+    Py_ssize_t encoded_size = PyBytes_Size(encoded_obj);
+    fwrite(encoded_str, 1, encoded_size, out);
+    Py_DECREF(encoded_obj);
+}
+
+static int
+find_line_number(PyCodeObject *code, _PyExecutorObject *executor)
+{
+    int code_len = (int)Py_SIZE(code);
+    for (int i = 0; i < code_len; i++) {
+        _Py_CODEUNIT *instr = &_PyCode_CODE(code)[i];
+        int opcode = instr->op.code;
+        if (opcode == ENTER_EXECUTOR) {
+            _PyExecutorObject *exec = code->co_executors->executors[instr->op.arg];
+            if (exec == executor) {
+                return PyCode_Addr2Line(code, i*2);
+            }
+        }
+        i += _PyOpcode_Caches[_Py_GetBaseCodeUnit(code, i).op.code];
+    }
+    return -1;
+}
+
+static void
+dump_executor(_PyExecutorObject *executor, FILE *out)
+{
+    PyCodeObject *code = executor->vm_data.code;
+    fprintf(out, "executor_%p [\n", executor);
+    fprintf(out, "    shape = none\n");
+    fprintf(out, "    label = <<table border=\"0\" cellspacing=\"0\">\n");
```

Comment:
```suggestion

    /* The body of the node is an HTML table with info about the executor.
       Any table row with a "port" attribute may be used later as the endpoint
       of a graph edge, identified by "executor_<id>:<port>".
       The first row has the "start" port, */
    fprintf(out, "    label = <<table border=\"0\" cellspacing=\"0\">\n");
```

Reply:
If we explain ports, why not cellspacing, border, etc?
I've added a link to the graphviz docs and examples, as that explains it better than I could.

Reply:
Ports are the only part in your PR that is relevant to relating nodes to edges.  Plus, ports are an unfamiliar concept while the others are not.  All that said, I suppose it isn't too hard to figure out if you think a comment isn't worth it.